opts/opts.go

```diff
@@ -515,3 +516,21 @@ func (m *MemSwapBytes) UnmarshalJSON(s []byte) error {
 	b := MemBytes(*m)
 	return b.UnmarshalJSON(s)
 }
+
+type OomScoreAdj int64
+
+func (o *OomScoreAdj) Type() string { return "int64" }
+
+func (o *OomScoreAdj) Value() int64 { return int64(*o) }
+
+func (o *OomScoreAdj) String() string {
+	return strconv.FormatInt(int64(*o), 10)
+}
+
+func (o *OomScoreAdj) Set(value string) error {
+
+	var conv int64
+	conv, _ = strconv.ParseInt(value, 10, 64)
+	*o = OomScoreAdj(conv)
```

Comment:
If `value` is not parseable as a number, the option gets silently set to 0? That doesn't seem right.

Reply:
I've run into silent defaults over mltiple code bases, if we want different/better behavior I didn't see anything documented.

Example...
https://github.com/docker/cli/blob/e174fe5597110959abd67e696802e239c6095931/opts/opts.go#L454

Reply:
> I've run into silent defaults over multiple code bases

What do "silent defaults" mean to you? Could you please elaborate?

> Example...

You have linked to the definition of the `String()` method of a [`pflag.Value`](https://pkg.go.dev/github.com/spf13/pflag#Value) implementation, which contains a documented special case for how its [zero value](https://go.dev/doc/effective_go#allocation_new:~:text=Since%20the%20memory,an%20unlocked%20mutex.) is formatted for user display.

https://github.com/docker/cli/blob/e174fe5597110959abd67e696802e239c6095931/opts/opts.go#L448-L450

I'm not following how this is relevant to the topic being discussed. Could you please explain?

Reply:
I consider silent defaults to be any configuration options that are:

**1.)**  set to values that have to be infered 
**2.)** not explicitly documented outside of source code

Reply:
I'm still lost. Why are we on this tangent about silent defaults? What do silent defaults have to do with a `(pflag.Value).Set()` method?

Reply:
There is no tangent, I'm just answering your question.

Reply:
Claiming that you have answered my question is not an answer. Please answer my question:

> What do silent defaults have to do with a `(pflag.Value).Set()` method?

Reply:
I didn't actually reference that, you did.

Reply:
`func (o *OomScoreAdj) Set(value string) error`, the body of which is the context for this discussion thread, implements a `Set` method necessary for `*OomScoreAdj` to satisfy the `pflag.Value` interface. In other words, `func (o *OomScoreAdj) Set(value string) error` is a `(pflag.Value).Set()` method.